### PR TITLE
[Submodule] Remove third-party onnx-tensorrt

### PR DESCRIPTION
**Pull request by cyyever**
It seems that tensorrt is not used by the C++ code, may be due to the removal of Caffe2.

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -1492,27 +1484,6 @@ if(CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO AND NOT INTERN_DISABLE_ONNX)
...
-  set(CMAKE_VERSION_ORIG "{CMAKE_VERSION}")
-  # TODO: this WAR is for https://github.com/pytorch/pytorch/issues/18524
-  set(CMAKE_VERSION "3.9.0")
-  add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/../third_party/onnx-tensorrt EXCLUDE_FROM_ALL)
-  set(CMAKE_VERSION "{CMAKE_VERSION_ORIG}")
-endfunction()
-if(CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
-  if(USE_TENSORRT)
-    add_onnx_tensorrt_subdir()
-    include_directories("${CMAKE_CURRENT_LIST_DIR}/../third_party/onnx-tensorrt")
```
Comment (Skylion007) [inline cmake/Dependencies.cmake:1509]: Can we remove the submodule too then?

File: CMakeLists.txt [hunk 2]
```diff
@@ -265,7 +265,6 @@ option(USE_PYTORCH_QNNPACK "Use ATen/QNNPACK (quantized 8-bit operators)" ON)
 option(USE_SNPE "Use Qualcomm's SNPE library" OFF)
 option(USE_SYSTEM_EIGEN_INSTALL
     "Use system Eigen instead of the one under third_party" OFF)
-option(USE_TENSORRT "Using Nvidia TensorRT library" OFF)
```
Comment (Skylion007) [inline CMakeLists.txt:268]: We also need to remove stuff from the bazel build system too, right?

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (cyyever) [reply cmake/Dependencies.cmake:1509]: This is what to do in the next PR.

File: CMakeLists.txt (hunk 2 above)
Comment (cyyever) [reply CMakeLists.txt:268]: Yes, I will remove bazel stuff and the submodule in the next PR.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**